app/src/main/java/com/humberto/tasky/agenda/presentation/AgendaViewModel.kt

```diff
@@ -20,37 +27,46 @@ import javax.inject.Inject
 class AgendaViewModel @Inject constructor(
     private val sessionManager: SessionManager,
     private val agendaRepository: AgendaRepository,
-    private val applicationScope: CoroutineScope
+    private val applicationScope: CoroutineScope,
+    private val localAgendaDataSource: LocalAgendaDataSource
 ): ViewModel() {
 
-    var state by mutableStateOf(AgendaState())
-        private set
+    private val _agendaState = MutableStateFlow(AgendaState())
+    val agendaState: StateFlow<AgendaState> = _agendaState.asStateFlow()
 
     private val eventChannel = Channel<AgendaEvent>()
     val events = eventChannel.receiveAsFlow()
 
     init {
         buildUserInitials()
+        getAgendaForDate(LocalDate.now())
     }
 
     private fun buildUserInitials() {
-        val fullName = sessionManager.get()?.fullName
-        fullName?.let {
-            state = state.copy(
-                initials = fullName.toInitials()
-            )
+        viewModelScope.launch {
+            val fullName = sessionManager.get()?.fullName
+            fullName?.let {
+                _agendaState.update { state ->
+                    state.copy(
+                        initials = fullName.toInitials()
+                    )
+                }
+            }
```

Comment:
Minor, you don’t seem to use the `fullName` variable, and it can be omitted.

```
sessionManager.get()?.let { session ->
  _agendaState.update { state ->
    state.copy(
      initials = session.fullName.toInitials()
    )
  }
}
```